platform/src/main/java/org/stellar/anchor/platform/paymentobserver/CirclePaymentObserverService.java

```diff
@@ -0,0 +1,217 @@
+  private final CirclePaymentObserverConfig circlePaymentObserverConfig;
+  private final String usdcIssuer;
+
+  private final Gson gson =
+      GsonUtils.builder()
+          .registerTypeAdapter(CircleTransfer.class, new CircleTransfer.Serialization())
+          .setPrettyPrinting()
+          .create();
+
+  public CirclePaymentObserverService(
+      OkHttpClient httpClient, CirclePaymentObserverConfig circlePaymentObserverConfig) {
+    this.httpClient = httpClient;
+    this.circlePaymentObserverConfig = circlePaymentObserverConfig;
+    Network stellarNetwork = StellarNetworkHelper.toStellarNetwork(circlePaymentObserverConfig.getStellarNetwork());
+    String[] assetIdPieces = CircleAsset.stellarUSDC(stellarNetwork).split(":");
+    this.usdcIssuer = assetIdPieces[assetIdPieces.length - 1];
+  }
+
+  public void handleCircleNotification(Map<String, Object> requestBody) {
+    CircleNotification circleNotification =
+        gson.fromJson(gson.toJson(requestBody), CircleNotification.class);
+    String type = circleNotification.getType();
+
+    switch (type) {
+      case "SubscriptionConfirmation":
+        handleSubscriptionConfirmationNotification(circleNotification);
+        return;
+
+      case "Notification":
+        handleTransferNotification(circleNotification);
+        return;
+
+      default:
+        Log.warn("Not handling notification of type " + type);
+    }
+  }
+
+  /**
+   * This will auto-subscribe to Circle when we receive a subscription available notification.
+   * @param circleNotification is the circle notification object.
+   */
+  public void handleSubscriptionConfirmationNotification(CircleNotification circleNotification) {
+    String subscribeUrl = circleNotification.getSubscribeUrl();
+    Log.info("=====> subscriptionNotification.subscribeUrl: " + subscribeUrl);
+
+    Request httpRequest =
+        new Request.Builder()
+            .url(subscribeUrl)
+            .header("Content-Type", "application/json")
+            .get()
+            .build();
+    Response response;
+    try {
+      response = httpClient.newCall(httpRequest).execute();
+    } catch (IOException e) {
+      Log.error("Failed to call endpoint " + subscribeUrl);
+      return;
+    }
+
+    Log.info(
+        String.format(
+            "Called subscribeUrl %s and got status code %d", subscribeUrl, response.code()));
+    if (response.code() != 200) {
+      Log.error(String.format("Status code: %d. \nResponse: %s", response.code(), response.body()));
+    }
+  }
+
+  /**
+   * Handle incoming circle notifications of type "transfers". A transfer notification can contain circle<>circle or circle<>stellar events.
+   * @param circleNotification is the circle notification object.
+   */
+  public void handleTransferNotification(CircleNotification circleNotification) {
+    TransferNotificationBody transferNotification =
+        gson.fromJson(circleNotification.getMessage(), TransferNotificationBody.class);
```

Comment:
Should we add a handler (controller or service) when message is mal-formed?

Reply:
🤔  how exactly would that be used?

Reply:
Something like in [Sep38Controller](https://github.com/stellar/java-stellar-anchor-sdk/blob/main/platform/src/main/java/org/stellar/anchor/platform/controller/Sep38Controller.java#L99)

Or we may wrap it and throw like:
```java
TransferNotificationBody transferNotification 
try {
transferNotification =
        gson.fromJson(circleNotification.getMessage(), TransferNotificationBody.class);
} catch (JsonParseException jpex) {
        throw new BadRequestException("description here...");
}
```

Reply:
Ok, I'll incorporate that in a future PR.